org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/javasupport/JdtQueuedBuildData.java

```diff
@@ -107,9 +108,11 @@ protected boolean doNeedRebuild(JavaBuilderState javaBuilderState, Procedure1<?
 				if (processor != null) {
 					processor.apply(ucDelta);
 				}
+			} else if (ucDelta.getProject().equals(project)) {
+				result = true;
```

Comment:
Should this also do a `ucDeltas.remove()`?

Reply:
No, it shouldn't.

Reply:
no, cause it is not confirmed yet here